notification/app/notification/services/fcm/AndroidConfigConverter.scala

```diff
@@ -0,0 +1,150 @@
+      case bn: BreakingNewsNotification => toBreakingNews(bn)
+      case ms: FootballMatchStatusNotification => toMatchStatusAlert(ms)
+    }
+  }
+
+
+  private def toAndroidTopic(topic: Topic) = s"${topic.`type`}//${topic.name}"
+
+  private def toBreakingNews(breakingNews: BreakingNewsNotification): FirebaseAndroidNotification = {
+
+    val sectionLink = condOpt(breakingNews.link) {
+      case Link.Internal(contentApiId, _, GITSection) => contentApiId
+    }
+
+    val tagLink = condOpt(breakingNews.link) {
+      case Link.Internal(contentApiId, _, GITTag) => contentApiId
+    }
+
+    val editions = breakingNews.topic
+      .filter(_.`type` == TopicTypes.Breaking)
+      .map(_.name)
+      .collect(Edition.fromString)
+
+    val androidLink = toAndroidLink(breakingNews.link)
+    val patformLink = toPlatformLink(breakingNews.link)
+    val edition = if (editions.size == 1) Some(editions.head) else None
+    val keyword = tagLink.map(new URI(_))
+
+    FirebaseAndroidNotification(
+      Map(
+        Keys.NotificationType -> AndroidMessageTypes.Custom,
+        Keys.UniqueIdentifier -> breakingNews.id.toString,
+        Keys.Type -> AndroidMessageTypes.Custom,
+        Keys.Title -> breakingNews.title,
+        Keys.Ticker -> breakingNews.message,
+        Keys.Message -> breakingNews.message,
+        Keys.Debug -> conf.debug.toString,
+        Keys.Editions -> editions.mkString(","),
+        Keys.Link -> toAndroidLink(breakingNews.link).toString,
+        Keys.UriType -> patformLink.`type`,
+        Keys.Uri -> patformLink.uri
+      ) ++ Map(
+        Keys.Section -> sectionLink.map(new URI(_)).map(_.toString),
+        Keys.Edition -> edition.map(_.toString),
+        Keys.Keyword -> keyword.map(_.toString),
+        Keys.ImageUrl -> breakingNews.imageUrl.map(_.toString),
+        Keys.ThumbnailUrl -> breakingNews.thumbnailUrl.map(_.toString)
+      ).flattenValues
+    )
+  }
+
+  private def toContent(cn: ContentNotification): FirebaseAndroidNotification = {
+    val link = toPlatformLink(cn.link)
+
+    FirebaseAndroidNotification(
+      Map(
+        Keys.Type -> AndroidMessageTypes.Custom,
+        Keys.UniqueIdentifier -> cn.id.toString,
+        Keys.Title -> cn.title,
+        Keys.Ticker -> cn.message,
+        Keys.Message -> cn.message,
+        Keys.Link -> toAndroidLink(cn.link).toString,
+        Keys.Topics -> cn.topic.map(toAndroidTopic).mkString(","),
+        Keys.UriType -> link.`type`,
+        Keys.Uri -> new URI(link.uri).toString,
+        Keys.Debug -> conf.debug.toString
+      ) ++ Map(
+        Keys.ThumbnailUrl -> cn.thumbnailUrl.map(_.toString)
+      ).flattenValues
+    )
+  }
+
+  private def toMatchStatusAlert(matchStatusAlert: FootballMatchStatusNotification): FirebaseAndroidNotification = FirebaseAndroidNotification(
+    Map(
+      "type" -> Some(AndroidMessageTypes.FootballMatchAlert),
+      "homeTeamName" -> Some(matchStatusAlert.homeTeamName),
```

Comment:
Is it easier to have a Map[String,String] ++ Map[String,Option[String]].flattenValues here too?

Reply:
You're right, I fixed